sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -229,19 +247,21 @@ export class ECEResponseProcessingService {
     disbursements: ECEDisbursements,
     auditUserId: number,
     processSummary: ProcessSummaryResult,
+    disbursementProcessingDetails: DisbursementProcessingDetails,
   ): Promise<DisbursementProcessingDetails> {
     const disbursementSchedules = Object.entries(disbursements);
-    const disbursementProcessingDetails = new DisbursementProcessingDetails();
     disbursementProcessingDetails.totalDisbursements =
       disbursementSchedules.length;
     for (const [
       disbursementScheduleId,
       disbursementDetails,
     ] of disbursementSchedules) {
-      const confirmedEnrolmentDetails = disbursementDetails.awardDetails.find(
-        (awardDetail) => awardDetail.isEnrolmentConfirmed,
-      );
       try {
+        this.validateDisbursement(disbursementDetails);
+        const confirmedEnrolmentDetails = disbursementDetails.awardDetails.find(
```

Comment:
this check is already happing in `validateDisbursement` can't we use that?

Reply:
This is not just a check, we find the confirmed enrolment and read it's information.

Again this logic is not something new and from the previous story(except the criteria that filters) #1501